include/deal.II/meshworker/mesh_loop.h

```diff
@@ -0,0 +1,286 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+#ifndef dealii__mesh_worker_mesh_loop_h
+#define dealii__mesh_worker_mesh_loop_h
+
+#include <deal.II/base/config.h>
+#include <deal.II/base/work_stream.h>
+#include <deal.II/base/template_constraints.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/filtered_iterator.h>
+#include <deal.II/meshworker/loop.h>
+#include <deal.II/meshworker/local_integrator.h>
+#include <deal.II/meshworker/dof_info.h>
+#include <deal.II/meshworker/integration_info.h>
+#include <deal.II/meshworker/assemble_flags.h>
+
+#include <functional>
+
+DEAL_II_NAMESPACE_OPEN
+
+template <typename> class TriaActiveIterator;
+
+namespace MeshWorker
+{
+  /**
+   * This function extends the WorkStream concept by externalising most of the
+   * work that is required to assemble face terms (for example in discontinuous
```

Comment:
... specializes WorkStream to operate on a mesh to do operations on cells, faces, and boundary faces.